test/org/zaproxy/zap/extension/pscanrules/CookieSecureFlagScannerUnitTest.java

```diff
@@ -0,0 +1,128 @@
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Set-Cookie: test=123; Path=/;\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("test"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("Set-Cookie: test"));
+    }
+
+    @Test
+    public void httpsSecureFlag () throws HttpMalformedHeaderException {
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Set-Cookie: test=123; Path=/; Secure;\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+    
+    @Test
+    public void secondCookieNoSecureFlag () throws HttpMalformedHeaderException {
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Set-Cookie: hasatt=test123; Path=/; Secure; HttpOnly\r\n" +
+                "Set-Cookie: test=123; Path=/;\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("test"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("Set-Cookie: test"));
+    }
+    
```

Comment:
Should we have a test case that sets multiple cookies in a single Set-Cookie header?

Something like:
`Set-Cookie: test=b;Path=/;,test2=d;Path=/;HttpOnly;Secure`